src/Illuminate/Database/Query/Builder.php

```diff
@@ -2247,7 +2247,7 @@ public function delete($id = null)
         // the ID to allow developers to simply and quickly remove a single row
         // from their database without manually specifying the where clauses.
         if (! is_null($id)) {
-            $this->where('id', '=', $id);
+            $this->where($this->from.'.id', '=', $id);
```

Comment:
Does this syntax cause any problems when you do a delete without a join? Does it care that you specify the table name in that case?